backend/src/controllers/course.js

```diff
@@ -1,24 +1,27 @@
-// TODO - MOVE TO SEPERATE FILE
-const qdb = require('../models/db-questions.js')
+const courseModel = require('../models/course')
+const questionModel = require('../models/question')
 
 /* Get specifc course data */
-exports.getCourse = function (req, res) {
-    const cid = req.params.cid
-    res.send(`<h1>Course Data for Course #${cid} </h1>`)
+exports.getCourse = function ({ params }, res) {
+    const courseID = params.id
+
+    courseModel.getCourse(courseID)
+        .then(data => res.json(data))
 }
 
 /* Get all questions for a course */
-exports.getCourseQuestions = function (req, res) {
-    const cid = req.params.cid
-    const pid = req.query.pid
+exports.getCourseQuestions = function ({ params, query }, res) {
+    const courseID = params.id
+    const pageNumber = query.p
 
-    qdb.getQuestions(cid, pid)
+    questionModel.getQuestions(courseID, pageNumber)
```

Comment:
I'd forgotten that if you pass a function signature to a then it will implicitly execute the function with the data you pas it. So. 

Any time you aren't doign a strict transform of the data in the then part and are just sending the json you can do this.

```js
doPromiseThing()
  .then(res.json)
```

Reply:
or even crazier. You could unpack json from the res object in the params and then do just.

```js
function({params, query}, {json}) {
    qdb.getQuestions()
       .then(json)
}
```

Reply:
Hmmm, couldn't get this to work.

Reply:
It should :-/